cpp-ch/local-engine/Parser/SerializedPlanParser.cpp

```diff
@@ -2117,7 +2117,14 @@ const ActionsDAG::Node * SerializedPlanParser::parseExpression(ActionsDAGPtr act
 QueryPlanPtr SerializedPlanParser::parse(const std::string & plan)
 {
     auto plan_ptr = std::make_unique<substrait::Plan>();
-    auto ok = plan_ptr->ParseFromString(plan);
+    /// https://stackoverflow.com/questions/52028583/getting-error-parsing-protobuf-data
+    /// Parsing may fail when the number of recursive layers is large.
+    /// Here, set a limit large enough to avoid this problem.
+    /// Once this problem occurs, it is difficult to troubleshoot, because the pb of c++ will not provide any valid information
+    google::protobuf::io::CodedInputStream coded_in(reinterpret_cast<const uint8_t *>(plan.data()), plan.size());
+    coded_in.SetRecursionLimit(100000);
```

Comment:
Is it enough for the most cases?

Reply:
With 16 threads on q4 accoured error. `100000` is enough for most cases, default is `64`.Need i place this in config file ?